p2p/ipld/plugin/nodes/nodes.go

```diff
@@ -207,6 +207,7 @@ func prependNode(newNode node.Node, nodes []node.Node) []node.Node {
 type NmtNodeAdder struct {
 	batch *format.Batch
 	ctx   context.Context
+	leaves *cid.Set
```

Comment:
Optional nit; reorder while at it:
```suggestion
	ctx   context.Context
	batch *format.Batch
	leaves *cid.Set
```